letsencrypt/scripts/main.py

```diff
@@ -9,61 +9,100 @@
+    add("-b", "--rollback", type=int, default=0, metavar="N",
+        help="Revert configuration N number of checkpoints.")
+    add("-v", "--view-config-changes", action="store_true",
+        help="View checkpoints and associated configuration changes.")
+    add("-r", "--redirect", action="store_true",
+        help="Automatically redirect all HTTP traffic to HTTPS for the newly "
+             "authenticated vhost.")
+
+    add("-e", "--agree-tos", dest="eula", action="store_true",
+        help="Skip the end user license agreement screen.")
+    add("-t", "--text", dest="use_curses", action="store_false",
+        help="Use the text output instead of the curses UI.")
+    add("--test", action="store_true", help="Run in test mode.")
+
+    # TODO: trailing slashes might be important! check and remove
+    add("--config-dir", default="/etc/letsencrypt/",
+        help="Configuration directory.")
+    add("--work-dir", default="/var/lib/letsencrypt/",
+        help="Working directory.")
+    add("--backup-dir", default="/var/lib/letsencrypt/backups/",
+        help="Configuration backups directory.")
+    add("--temp-checkpoint-dir",
+        default="/var/lib/letsencrypt/temp_checkpoint/",
+        help="Temporary checkpoint directory.")
+    add("--in-progress-dir",
+        default="/var/lib/letsencrypt/backups/IN_PROGRESS/",
+        help="Directory used before a permanent checkpoint is finalized")
+    add("--cert-key-backup", default="/var/lib/letsencrypt/keys-certs/",
+        help="Directory where all certificates and keys are stored. "
+             "Used for easy revocation.")
+    add("--rev-tokens-dir", default="/var/lib/letsencrypt/revocation_tokens/",
+        help="Directory where all revocation tokens are saved.")
+    add("--key-dir", default="/etc/letsencrypt/keys/", help="Keys storage.")
+    add("--cert-dir", default="/etc/letsencrypt/certs/",
+        help="Certificates storage.")
+
+    add("--le-vhost-ext", default="-le-ssl.conf",
+        help="SSL vhost configuration extension.")
+    add("--cert-path", default="/etc/letsencrypt/certs/cert-letsencrypt.pem",
+        help="Let's Encrypt certificate file.")
+    add("--chain-path", default="/etc/letsencrypt/certs/chain-letsencrypt.pem",
+        help="Let's Encrypt chain file.")
+
+    add("--apache-ctl", default="apache2ctl",
+        help="Path to the 'apache2ctl' binary, used for 'configtest' and "
+             "retrieving Apache2 version number.")
+    add("--apache-enmod", default="a2enmod",
+        help="Path to the Apache 'a2enmod' binary.")
+    add("--apache-init-script", default="/etc/init.d/apache2",
+        help="Path to the Apache init script (used for server reload/restart).")
+    add("--apache-server-root", default="/etc/apache2",
+        help="Apache server root directory.")
+    add("--apache-mod-ssl-conf", default="/etc/letsencrypt/options-ssl.conf",
+        help="Contains standard Apache SSL directives.")
+
```

Comment:
1.  I have never said I expected users to supply paths with a trailing slash...  I have actually mentioned multiple times that it is a TODO item and it was locked away in a config file.

2.  You brought it out to the command-line... that is fine, but either keep the code working... by copying and pasting the previous code and keep the TODO item or fix it completely.  I gave you the option of choosing the shortest route to completion instead of making a partial fix to class that is going to be revised.